orbit/pkg/update/swift_dialog.go

```diff
@@ -34,10 +34,12 @@ func (s *SwiftDialogDownloader) Run(cfg *fleet.OrbitConfig) error {
 		return nil
 	}
 
-	if !cfg.Notifications.NeedsMDMMigration && !cfg.Notifications.RenewEnrollmentProfile {
-		log.Debug().Msg("got false needs migration and false renew enrollment")
-		return nil
-	}
+	// // TODO: we probably want to ensure that swiftDialog is always installed if we're going to be
+	// // using it offline.
+	// if !cfg.Notifications.NeedsMDMMigration && !cfg.Notifications.RenewEnrollmentProfile {
+	// 	log.Debug().Msg("got false needs migration and false renew enrollment")
+	// 	return nil
+	// }
```

Comment:
does this mean we'll download swiftDialog for everyone? IMO it's important to keep in mind the main problem we're trying to solve: users losing connectivity _on unenrollment_ (because profiles are removed.)

In this world, swiftDialog will be there (that's how you start the migration) and we should keep this check.

Reply:
> In this world, swiftDialog will be there (that's how you start the migration) and we should keep this check.

Are we sure about that? What happens if Orbit/Fleet Desktop restarts and goes offline? In that case, I think the runner won't register the target because it doesn't get `cfg.Notifications`, right?

Reply:
and you're thinking Fleet Desktop depends on the target being registered?

Reply:
do you mind pointing me where? I think we should fix that instead

Reply:
Hmm, I'm seeing behavior perhaps even worse than that. It seems if Orbit is killed while the device is offline, Fleet Desktop will never relaunch until the device comes back online.

Reply:
> Fleet Desktop will never relaunch until the device comes back online

@gillespi314 is that because of the `deviceEnabledChan` that blocks all the other goroutines?

Reply:
@jahzielv, I think it is happening in Orbit itself. I don't even see logs from Desktop that I would expect if it were the deviceEnabled channel issue. When you have a second, would you mind giving it a try locally and let me know what you see? Just turn off wifi, pkill orbit and watch the logs for both Orbit and Desktop?

Reply:
@gillespi314 yeah, orbit logs show only failed attempts to talk to the TUF server and to Fleet, and Fleet Desktop never comes back up.

Reply:
that kind of makes sense, given how orbit works today. Is it worth bringing to the team at large? changing this behavior might skew the estimate.

Reply:
Yeah worth bringing up at least @roperzh

Reply:
I'll add to the parking lot for today's standup.